pkg/serviceaccount/claims.go

```diff
@@ -212,12 +273,19 @@ func (v *validator) Validate(ctx context.Context, _ string, public *jwt.Claims,
 		}
 	}
 
+	var jti string
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenJTI) {
+		jti = public.ID
+	}
 	return &apiserverserviceaccount.ServiceAccountInfo{
-		Namespace: private.Kubernetes.Namespace,
-		Name:      private.Kubernetes.Svcacct.Name,
-		UID:       private.Kubernetes.Svcacct.UID,
-		PodName:   podName,
-		PodUID:    podUID,
+		Namespace:    private.Kubernetes.Namespace,
+		Name:         private.Kubernetes.Svcacct.Name,
+		UID:          private.Kubernetes.Svcacct.UID,
+		PodName:      podName,
+		PodUID:       podUID,
+		NodeName:     nodeName,
+		NodeUID:      nodeUID,
+		CredentialID: apiserverserviceaccount.CredentialIDForJTI(jti),
```

Comment:
this is failing the integration test because it always outputs "JTI=" even when jti is ""

make that conditional